src/Stratis.Bitcoin/Consensus/ConsensusManagerBehavior.cs

```diff
@@ -454,11 +449,25 @@ protected async Task<ConnectNewHeadersResult> PresentHeadersLockedAsync(List<Blo
             }
             catch (ConnectHeaderException)
             {
+                // This is typically thrown when the first header refers to a previous block hash that is not in
+                // the header tree currently. This is not regarded as bannable, as it could be a legitimate reorg.
                 this.logger.LogDebug("Unable to connect headers.");
-                this.cachedHeaders.Clear();
 
-                // Resync in case can't connect.
-                await this.ResyncAsync().ConfigureAwait(false);
+                if (this.CheckIfUnsolicitedFutureHeader(headers))
+                {
+                    // However, during IBD it is much more likely that the unconnectable header is an unsolicited
+                    // block advertisement. In that case we just ignore the failed header and don't modify the cache.
+                    // TODO: Review more closely what Bitcoin Core does here - they seem to allow 8 unconnectable headers before
+                    // applying partial DoS points to a peer. Incorporate that into rate limiting code?
+                    this.logger.LogTrace("(-)[HEADER_FUTURE_CANT_CONNECT]");
```

Comment:
I suggest this exit trace to be different then the other one, in case it appears in logs it will be easy to distinguish where it happened.

Reply:
Done, not a particularly imaginative rename but I changed it.